target/linux/generic/files/drivers/net/phy/rtl8367b.c

```diff
@@ -730,35 +731,23 @@ static int rtl8367b_write_phy_reg(struct rtl8366_smi *smi,
 static int rtl8367b_init_regs(struct rtl8366_smi *smi)
 {
 	const struct rtl8367b_initval *initvals;
-	u32 chip_num;
-	u32 chip_ver;
-	u32 rlvid;
 	int count;
-	int err;
-
-	REG_WR(smi, RTL8367B_RTL_MAGIC_ID_REG, RTL8367B_RTL_MAGIC_ID_VAL);
-	REG_RD(smi, RTL8367B_CHIP_NUMBER_REG, &chip_num);
-	REG_RD(smi, RTL8367B_CHIP_VER_REG, &chip_ver);
 
-	if ((chip_ver ==  0x0020 || chip_ver == 0x00A0) && chip_num == 0x6367)  {
+	switch (smi->rtl8367b_chip) {
+	case RTL8367B_CHIP_UNKNOWN_RLVID0:
+		initvals = rtl8367r_vb_initvals_0;
+		count = ARRAY_SIZE(rtl8367r_vb_initvals_0);
+		break;
+	case RTL8367B_CHIP_UNKNOWN_RLVID1 ... RTL8367B_CHIP_RTL8367RB:
+		initvals = rtl8367r_vb_initvals_1;
+		count = ARRAY_SIZE(rtl8367r_vb_initvals_1);
+		break;
+	case RTL8367B_CHIP_RTL8367S ... RTL8367B_CHIP_RTL8367RB_VB:
```

Comment:
I do not like the `...` in the case statement, otherwise this change looks fine.
Please mention all options so it does not break in case someone adds a new enum option in the middle.

Reply:
"..." removed

Reply:
"..." syntax removed
The default case is basically run for RTL8367B_CHIP_UNKNOWN